crates/router/src/connector/noon.rs

```diff
@@ -260,7 +274,59 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
         req: &types::PaymentsAuthorizeRouterData,
         _connectors: &settings::Connectors,
     ) -> CustomResult<RequestContent, errors::ConnectorError> {
-        let connector_req = noon::NoonPaymentsRequest::try_from(req)?;
+        let amount = connector_utils::convert_amount(
+            self.amount_converter,
+            req.request.minor_amount,
+            req.request.currency,
+        )?;
+
+        // The description should not have leading or trailing whitespaces, also it should not have double whitespaces and a max 50 chars according to Noon's Docs
+        let name: String = req
+            .get_description()?
+            .trim()
+            .replace("  ", " ")
+            .chars()
+            .take(50)
+            .collect();
+
+        let subscription: Option<noon::NoonSubscriptionData> = req
+            .request
+            .get_setup_mandate_details()
+            .map(|mandate_data| {
+                let max_amount = match &mandate_data.mandate_type {
+                    Some(hyperswitch_domain_models::mandates::MandateDataType::SingleUse(
+                        mandate,
+                    ))
+                    | Some(hyperswitch_domain_models::mandates::MandateDataType::MultiUse(Some(
+                        mandate,
+                    ))) => connector_utils::convert_amount(
+                        self.amount_converter,
+                        mandate.amount,
+                        mandate.currency,
+                    ),
+                    Some(hyperswitch_domain_models::mandates::MandateDataType::MultiUse(None)) => {
+                        Err(errors::ConnectorError::MissingRequiredField {
+                            field_name:
+                                "setup_future_usage.mandate_data.mandate_type.multi_use.amount",
+                        }
+                        .into())
+                    }
+                    None => Err(errors::ConnectorError::MissingRequiredField {
+                        field_name: "setup_future_usage.mandate_data.mandate_type",
+                    }
+                    .into()),
+                }?;
```

Comment:
Transformer logic should not be in connector mod files, we need to solutionize for a way where converting capabilities are either available at transformer level or a way to pass all the amounts as arguments to try_froms

Reply:
done, moved transformer logic to transformer file.